[OPIK-532] Return detailed error message for json deserialization exceptions

BorisTkachenko opened the pull request:
## Details
Return detailed error message for json deserialization exceptions

## Testing
Added corresponding integration test

thiagohora commented on apps/opik-backend/src/main/java/com/comet/opik/api/error/JsonInvalidFormatExceptionMapper.java:
```diff
@@ -0,0 +1,16 @@
+
+    @Override
+    public Response toResponse(InvalidFormatException exception) {
+        log.warn("Deserialization exception: {}", exception.getMessage());
```
We must carefully to avoid a few details related to the framework or infrastructure. Can we create a

andrescrz replied:
Minor: this is caused by bad request, no need to log as warning level, probably info is fine. Only if many infos might be an indicator of something broken on our side. Otherwise, most likely users just making mistakes.

thiagohora commented on apps/opik-backend/src/main/java/com/comet/opik/api/error/JsonInvalidFormatExceptionMapper.java:
```diff
@@ -0,0 +1,16 @@
+    @Override
+    public Response toResponse(InvalidFormatException exception) {
+        log.warn("Deserialization exception: {}", exception.getMessage());
+        return Response.status(Response.Status.BAD_REQUEST).entity(exception.getMessage()).build();
```
We must carefully to avoid a few details related to the framework or infrastructure. I believe the best would be to create a dedicated serializer for your use case.

BorisTkachenko replied:
This error message in response has nothing to do with infrastructure. And regarding the framework, it's an open source anyway, so I'm not sure why we can't send a detailed message.

Dedicated serializer will not help here. Main pain point is debugging, now returned error swallows all details and not informative. It's just `Unable to process JSON`. Mistake could be made in any field, we can't predict all possible scenarios and prepare serializers for it.

andrescrz replied:
We just need to be careful with our exception messages in order to avoid issues here. Otherwise, we're mostly fine.

andrescrz replied:
Additinally, I recommend you using this`io.dropwizard.jersey.errors.ErrorMessage` class instead of String as entity. That's the standard error format, otherwise, our endpoints won't be consistent.

thiagohora replied:
Yeah, the point is that the error maybe too technical and not helpful at all

andrescrz replied:
In that case, maybe Boris can adjust this to a custom message, which is more informative. But he should at least parse the exception message to know which field is the problematic one. It might be tricker than just dumping the whole message.

BorisTkachenko commented:
@andrescrz @thiagohora I addressed your comments. Also regarding the format of error message discussed with Jacques and updated accordingly: https://comet-ml.slack.com/archives/C0768KH9MDJ/p1733403958715779
